match2/commons/opponent_display.lua

```diff
@@ -212,33 +235,56 @@ OpponentDisplay.propTypes.BlockTeam = {
 
 --[[
 Displays a team as a block element. The width of the component is determined by
-its layout context, and not of the team name.
+its layout context, and not of the team name. The team is specified by template.
 ]]
-function OpponentDisplay.BlockTeam(props)
-	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.BlockTeam)
-	local style = props.style or 'standard'
+function OpponentDisplay.BlockTeamContainer(props)
+	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.BlockTeamContainer)
 
-	local raw = mw.ext.TeamTemplate.raw(props.template)
-	if not raw then
+	local team = MatchGroupUtil.fetchTeam(props.template)
+	if not team then
 		return mw.html.create('div'):addClass('error')
 			:wikitext('No team template exists for name ' .. props.template)
 	end
 
-	local displayName = style == 'standard' and raw.name
-		or style == 'short' and raw.shortname
-		or style == 'bracket' and raw.bracketname
+	return OpponentDisplay.BlockTeam(Table.merge(props, {
+		icon = mw.ext.TeamTemplate.teamicon(props.template),
+		team = team,
+	}))
+end
+
+OpponentDisplay.propTypes.BlockTeam = {
+	flip = 'boolean?',
+	icon = 'any',
+	overflow = TypeUtil.optional(DisplayUtil.types.OverflowModes),
+	showLink = 'boolean?',
+	style = TypeUtil.optional(OpponentDisplay.types.TeamStyle),
+	team = MatchGroupUtil.types.Team,
+}
+
+--[[
+Displays a team as a block element. The width of the component is determined by
+its layout context, and not of the team name. The team is specified by a team
+struct and icon wikitext.
+]]
+function OpponentDisplay.BlockTeam(props)
+	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.BlockTeam)
+	local style = props.style or 'standard'
+
+	local displayName = style == 'standard' and props.team.displayName
+		or style == 'short' and props.team.shortName
+		or style == 'bracket' and props.team.bracketName
```

Comment:
These strings could benefit from constants to make this a little easier to understand

Reply:
If you mean putting string values in variables and referring to the variables instead
```
local standard = 'standard'
local displayName = style == standard and ...
```

I generally prefer to use the string literals directly.